src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -242,6 +243,13 @@ public WorkspaceDescription createWorkspace(
                   CREATE_WORKSPACE_MAXIMUM_RETRIES,
                   CREATE_WORKSPACE_DURATION_SLEEP_FOR_RETRY);
           logger.debug("create workspace context result: {}", createContextResult);
+          StatusEnum status = createContextResult.getJobReport().getStatus();
+          if (StatusEnum.FAILED == status || StatusEnum.RUNNING == status) {
```

Comment:
Just to confirm: we're polling the create context call above, so if it's still in `RUNNING` status at this point we consider it failed?

Reply:
That's what `throwIfJobNotCompleted` does right below. `"CLI timed out waiting for the job to complete. It's still running on the server."`

Reply:
I think we should only delete the workspace if the context creation fails, not if it times out. 2 reasons:
- If e.g. RBS gets overloaded and creating a context takes a long time, then we'd have no way of creating workspaces. If instead we just let the job keep running, the user will get a timed out message but can still use the workspace after the job completes.
- The main use case for this change at the moment is when a user has no spend profile access. That's an early step in the create context flight, so I wouldn't expect timeouts.

Longer term, we need to add CLI commands for managing long-running jobs (PF-745), including some kind of callback on success/failure. That would cover the case when the job times out and ultimately fails. For now, let's assume that if the job times out it will ultimately succeed, because I think that will be more common.

Reply:
done